Correctly raise ``SyntaxError`` on exception groups (:pep:`654`) on python
versions prior to 3.11
